src/layers/conv.jl

```diff
@@ -680,10 +680,17 @@ Flux.trainable(l::GINConv) = (nn = l.nn,)
 
 GINConv(nn, ϵ; aggr = +) = GINConv(nn, ϵ, aggr)
 
-function (l::GINConv)(g::GNNGraph, x::AbstractMatrix)
+function (l::GINConv)(g::AbstractGNNGraph, x)
     check_num_nodes(g, x)
-    m = propagate(copy_xj, g, l.aggr, xj = x)
-    l.nn((1 + ofeltype(x, l.ϵ)) * x + m)
+    xj, xi = expand_srcdst(g, x) 
+ 
+    m = propagate(copy_xj, g, l.aggr, xj = xj)
+    
+    if g isa GNNHeteroGraph
+        l.nn((1 .+ ofeltype(xi, l.ϵ)) .* xi .+ m)
+    else
+        l.nn((1 + ofeltype(x, l.ϵ)) * x + m)
+    end
```

Comment:
why there is an `if` here instead of just using `xi`?

Reply:
fixed